src/main/java/org/aksw/simba/lemming/metrics/single/SingleValueMetric.java

```diff
@@ -1,22 +1,61 @@
-package org.aksw.simba.lemming.metrics.single;
-
-import org.aksw.simba.lemming.ColouredGraph;
-import org.aksw.simba.lemming.metrics.Metric;
-
-/**
- * A metric that generates a single double value.
- * 
- * @author Michael R&ouml;der (roeder@informatik.uni-leipzig.de)
- *
- */
-public interface SingleValueMetric extends Metric {
-
-    /**
-     * Applies the metric to the given graph.
-     * 
-     * @param graph
-     *            the graph for which the metric should be calculated.
-     * @return the value of the metric.
-     */
-    public double apply(ColouredGraph graph);
-}
+package org.aksw.simba.lemming.metrics.single;
+
+import org.aksw.simba.lemming.ColouredGraph;
+import org.aksw.simba.lemming.metrics.Metric;
+import org.aksw.simba.lemming.metrics.single.edgemanipulation.VertexDegrees;
+import org.aksw.simba.lemming.mimicgraph.constraints.TripleBaseSingleID;
+
+/**
+ * A metric that generates a single double value.
+ * 
+ * @author Michael R&ouml;der (roeder@informatik.uni-leipzig.de)
+ *
+ */
+public interface SingleValueMetric extends Metric {
+
+    /**
+     * Applies the metric to the given graph.
+     * 
+     * @param graph
+     *            the graph for which the metric should be calculated.
+     * @return the value of the metric.
+     */
+    public double apply(ColouredGraph graph);
+    
+    
+	/**
+	 * Returns metric results that for further computations could be reused.
+	 * 
+	 * @param graph
+	 *            - input graph.
+	 * @param prevMetricResult
+	 *            - UpdatableMetricResult object containing the results that could
+	 *            be used.
+	 * @return - metric result.
+	 */
+	public default UpdatableMetricResult applyUpdatable(ColouredGraph graph, UpdatableMetricResult prevMetricResult) {
```

Comment:
The signature of this method is still wrong. It means that for creating an instance of an `UpdatableMetricResult` another `UpdatableMetricResult` is needed. However, that is not the case. This method should simply be the `apply` method with a different return type. You can also see that from the default implementation.

Reply:
Done. I have removed the UpdatableMetricResult parameter.